quodlibet/util/collection.py

```diff
@@ -47,19 +48,73 @@
 ELPOEP = list(reversed(PEOPLE))
 PEOPLE_SCORE = [100 ** i for i in range(len(PEOPLE))]
 
+SMOOTHING_FACTOR_DEFAULT = 10
 
-def avg(nums):
+
+def unweighted_average(nums):
     """Returns the average (arithmetic mean) of a list of numbers"""
     return float(sum(nums)) / len(nums)
 
 
-def bayesian_average(nums, c=None, m=None):
+def unweighted_bayesian_average(nums, c=None, m=None):
     """Returns the Bayesian average of an iterable of numbers,
     with parameters defaulting to config specific to ~#rating."""
     m = m or config.RATINGS.default
     c = c or config.getfloat("settings", "bayesian_rating_factor", 0.0)
-    ret = float(m * c + sum(nums)) / (c + len(nums))
-    return ret
+    return float(m * c + sum(nums)) / (c + len(nums))
+
+
+def hyperbolic_smoother(x, n=SMOOTHING_FACTOR_DEFAULT):
+    """n determines how strongly smaller values will be magnified"""
+    return (n + 1) * x / (n * x + 1)
+
+
+def smoothed_length_weights_and_nums(songs, key, total_length, smoothing_factor):
+    """Returns smoothed length-based weights and the songs' associated numeric values"""
+    length_based_smoothed_weights = []
+    values = []
+    for s in songs:
+        value = s(key)
+        if value == "":
```

Comment:
```suggestion
        if not value:
```
Unless there's some good reason (in which case a comment would be good!)

Reply:
This check comes from here:

https://github.com/quodlibet/quodlibet/blob/f1bd2739c77d774aa42664aa1897e08456939705/quodlibet/util/collection.py#L298

So, I'll add the comment from there:

```python
 	                # If none of the songs can return a numeric key,
	                # the album returns default
```

Reply:
Actually, I'll try to add another comment that seems clearer